api/src/main/java/com/okta/sdk/resource/policy/PolicyBuilder.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017 Okta
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.sdk.resource.policy;
+
+import com.okta.commons.lang.Classes;
+import com.okta.sdk.client.Client;
+
+
+public interface PolicyBuilder {
```

Comment:
This should probably be something like `PolicyBuilder<T extends PolicyBuilder>` 
and then all of the methods should return `T`.

And the corresponding above `PasswordPolicyBuilder` would be:
```java
public interface PasswordPolicyBuilder extends PolicyBuilder<PasswordPolicyBuilder> { ... }
```

In the `PolicyBuilder` you would likely need to cast to `(T)`, and to avoid all of the cast warnings you can do something like this: https://github.com/paseto-toolkit/jpaseto/blob/master/impl/src/main/java/dev/paseto/jpaseto/impl/AbstractPasetoBuilder.java#L53-L59

Reply:
Adding generic will break DefaultPasswordPolicyBuilder.class invocation from within the PasswordPolicyBuilder interface. How do we go around that in this case? Please advise.

Reply:
Sorry, I don't understand the question.
NOTE: my suggestions were from memory, so I very well could have missed something

Reply:
No problem at all, when we add generics as suggested up here, we below instantiation of the implementation from the `PasswordPolicyBuilder` will fail. So the question was, how do we get around it.

` static PasswordPolicyBuilder instance() {
        return Classes.newInstance("com.okta.sdk.impl.resource.DefaultPasswordPolicyBuilder");
    }`

Reply:
What is the runtime exception?

Reply:
com.okta.commons.lang.InstantiationException: Unable to instantiate class [com.okta.sdk.impl.resource.DefaultPasswordPolicyBuilder]

Reply:
Awesome! Thank you!